vllm/model_executor/models/registry.py

```diff
@@ -94,33 +94,23 @@
 _EMBEDDING_MODELS = {
     # [Text-only]
     "BertModel": ("bert", "BertEmbeddingModel"),
+    "DeciLMForCausalLM": ("decilm", "DeciLMForCausalLM"),
     "Gemma2Model": ("gemma2", "Gemma2EmbeddingModel"),
     "LlamaModel": ("llama", "LlamaEmbeddingModel"),
+    **{
+        # Multiple models share the same architecture, so we include them all
+        k: (mod, arch) for k, (mod, arch) in _TEXT_GENERATION_MODELS.items()
+        if arch == "LlamaForCausalLM"
+    },
```

Comment:
The PR just mentions sorting the list, though this appears to add new elements to the list, as well. I assume that's intentional, but wanted to verify ...

Reply:
These models were previously added via `add_embedding_models`. This dictionary comprehension replaces it.

Reply:
Oh, I see, thanks!